archive/100-theorems-list/90_stirlings_formula.lean

```diff
@@ -0,0 +1,233 @@
+    { rw [cast_succ, one_div],
+      refine inv_le_one _,
+      norm_cast,
+      exact (le_add_iff_nonneg_left 1).mpr zero_le', },
+    exact mul_le_of_le_one_left h_zero_le h_left, },
+  exact has_sum_le hab (log_stirling_seq_diff_has_sum n) g,
+end
+
+/--
+We have the bound  `log_stirling_seq n - log_stirling_seq (n+1)` ≤ 1/(4 n^2)
+-/
+lemma log_stirling_seq_sub_log_stirling_seq_succ (n : ℕ) :
+  log_stirling_seq n.succ - log_stirling_seq n.succ.succ ≤ 1 / (4 * n.succ ^ 2) :=
+begin
+  have h₁ : 0 < 4 * ((n : ℝ) + 1) ^ 2 := by nlinarith [@cast_nonneg ℝ _ n],
+  have h₃ : 0 < (2 * ((n : ℝ) + 1) + 1) ^ 2 := by nlinarith [@cast_nonneg ℝ _ n ],
+  have h₂ : 0 < 1 - (1 / (2 * ((n : ℝ) + 1) + 1)) ^ 2,
+  { rw ← mul_lt_mul_right h₃,
+    have H : 0 < (2 * ((n : ℝ) + 1) + 1) ^ 2 - 1 := by nlinarith [@cast_nonneg ℝ _ n ],
+    convert H using 1; field_simp [h₃.ne'] },
+  refine le_trans (log_stirling_seq_diff_le_geo_sum n) _,
+  push_cast at *,
+  rw div_le_div_iff h₂ h₁,
+  field_simp [h₃.ne'],
+  rw div_le_div_right h₃,
+  ring_nf,
+  norm_cast,
+  linarith,
+end
+
+/-- For any `n`, we have `log_stirling_seq 1 - log_stirling_seq n ≤ 1/4 * ∑' 1/k^2`  -/
+lemma log_stirling_seq_bounded_aux : ∃ (c : ℝ), ∀ (n : ℕ),
+  log_stirling_seq 1 - log_stirling_seq n.succ ≤ c :=
+begin
+  let d := ∑' k : ℕ, (1 : ℝ) / k.succ ^ 2,
+  use (1 / 4 * d : ℝ),
+  let log_stirling_seq' : ℕ → ℝ := λ k : ℕ, log_stirling_seq k.succ,
+  intro n,
+  calc
+  log_stirling_seq 1 - log_stirling_seq n.succ = log_stirling_seq' 0 - log_stirling_seq' n : rfl
+    ... = ∑ k in range n, (log_stirling_seq' k - log_stirling_seq' (k + 1)) : by
+      rw ← sum_range_sub' log_stirling_seq' n
+    ... ≤ ∑ k in range n, (1/4) * (1 / k.succ^2) : by
+    { apply sum_le_sum,
+      intros k hk,
+      convert log_stirling_seq_sub_log_stirling_seq_succ k using 1,
+      field_simp, }
+    ... = 1 / 4 * ∑ k in range n, 1 / k.succ ^ 2 : by rw mul_sum
+    ... ≤ 1 / 4 * d : by
+    { refine (mul_le_mul_left _).mpr _, { exact one_div_pos.mpr four_pos, },
+      refine sum_le_tsum (range n) (λ k _, _)
+        ((summable_nat_add_iff 1).mpr (real.summable_one_div_nat_pow.mpr one_lt_two)),
+      apply le_of_lt,
+      rw one_div_pos,
+      rw sq_pos_iff,
+      exact nonzero_of_invertible ↑(succ k), },
+end
+
+/-- The sequence `log_stirling_seq` is bounded below for `n ≥ 1`. -/
+lemma log_stirling_seq_bounded_by_constant : ∃ c, ∀ (n : ℕ), c ≤ log_stirling_seq n.succ :=
+begin
+  obtain ⟨d, h⟩ := log_stirling_seq_bounded_aux,
+  use log_stirling_seq 1 - d,
+  intro n,
+  exact sub_le.mp (h n),
+end
+
+/-- The sequence `stirling_seq` is positive for `n > 0`  -/
+lemma stirling_seq'_pos (n : ℕ) : 0 < stirling_seq n.succ :=
+begin
+  apply_rules [cast_pos.mpr, factorial_pos, exp_pos, pow_pos, div_pos, mul_pos, real.sqrt_pos.mpr,
+    two_pos, succ_pos];
+  apply_instance
+end
+
+/--
+The sequence `stirling_seq` has a positive lower bound (in fact, `exp (3/4 - 1/2 * log 2)`)
+-/
```

Comment:
Well, in fact, the best positive lower bound is `sqrt π`, as you prove in your next PR.  You could make this sentence accurate by saying that the lower bound `exp (3/4 - 1/2 * log 2)` is relatively easy to establish by an extension of the technique in this lemma, or you could omit the "in fact" comment altogether.

Reply:
Yeah, this was leftover from before we changed the proof to use a different constant. 
Removing it, good catch!